build-service.py

```diff
@@ -179,6 +181,47 @@ def build_repo(repo_dir, branch, deploy_conf, service_conf):
     return image_name
 
 
+# create a new namespace specified in the deployment script and the namespace is not already in the system.
+# otherwise, if not specified, use default as namespace
+def create_namespace(kubernetes_api, deploy_conf):
+
+    if "namespace" not in deploy_conf["metadata"]:
+        # namespace is not specified in deplyoment config, just set namespace as default
+        deploy_conf["metadata"]["namespace"] = "default"
+        return
+    elif deploy_conf["metadata"]["namespace"] == "default":
+        # do nothing
+        return
+
+    ns = deploy_conf["metadata"]["namespace"]
+    try:
+        namespaces = kubernetes_api.list_namespaced_deployment(ns)
+        if len(namespaces.items) > 0:
+            logging.debug(f"deployments found for {ns}. Don't need to make namespace")
+        else:
+            try:
+                testclient = dynamic.DynamicClient(
+                    api_client.ApiClient(configuration=config.load_kube_config())
+                )
+                namespace_api = testclient.resources.get(
+                    api_version="v1", kind="Namespace"
+                )
+                namespace_manifest = {
+                    "apiVersion": "v1",
+                    "kind": "Namespace",
+                    "metadata": {"name": ns, "resourceversion": "v1"},
+                }
+                namespace_api.create(body=namespace_manifest)
+                logging.debug(f"created a new namespace :{namespace_manifest}")
+            except Exception as e:
+                logging.error(f"failed to create namespace: {e}")
+                return {"err": f"Failed to create namespace: {e}"}, 500
```

Comment:
This does not work for the CLI -- youll need to create a custom exception like I did for errors in `build_repo()` so that the daemon and CLI can handle them correctly for their interfaces